rotkehlchen/db/snapshots.py

```diff
@@ -0,0 +1,108 @@
+        self.snapshot_archive_path = self.tmp_dir_path / 'snapshot.zip'
+        self.timed_balances_csv_path = self.tmp_dir_path / 'timed_balances.csv'
+        self.timed_location_data_csv_path = self.tmp_dir_path / 'timed_location_data.csv'
+
+    def get_timed_balances(self, timestamp: Timestamp) -> List[Dict[str, Any]]:
+        balances_data = []
+        cursor = self.db_handler.conn.cursor()
+        timed_balances_result = cursor.execute(
+            'SELECT category, time, amount, currency, usd_value FROM timed_balances '
+            'WHERE time=?', (timestamp,),
+        )
+        for data in timed_balances_result:
+            balances_data.append(
+                DBAssetBalance(
+                    category=BalanceType.deserialize_from_db(data[0]),
+                    time=data[1],
+                    amount=data[2],
+                    asset=data[3],
+                    usd_value=data[4],
+                ),
+            )
+        return self.clean_data(balances_data)
+
+    def get_timed_location_data(self, timestamp: Timestamp) -> List[Dict[str, Any]]:
+        location_data = []
+        cursor = self.db_handler.conn.cursor()
+        timed_location_data = cursor.execute(
+            'SELECT time, location, usd_value FROM timed_location_data '
+            'WHERE time=?',
+            (timestamp,),
+        )
+        for data in timed_location_data:
+            location_data.append(
+                LocationData(
+                    time=data[0],
+                    location=data[1],
+                    usd_value=data[2],
+                ),
+            )
+
+        return self.clean_data(location_data)
+
+    def export(
+        self,
+        timed_balances_data: List[Dict[str, Any]],
+        timed_location_data: List[Dict[str, Any]],
+    ) -> Tuple[bool, str]:
+        try:
+            with open(self.timed_balances_csv_path, 'w') as f:
+                w = csv.DictWriter(f, timed_balances_data[0].keys())
+                w.writeheader()
+                w.writerows(timed_balances_data)
+
+            with open(self.timed_location_data_csv_path, 'w') as f:
+                w = csv.DictWriter(f, timed_location_data[0].keys())
+                w.writeheader()
+                w.writerows(timed_location_data)
+
+            with ZipFile(self.snapshot_archive_path, mode='w') as archive:
+                archive.write(self.timed_balances_csv_path)
+                archive.write(self.timed_location_data_csv_path)
+
+            return True, str(self.snapshot_archive_path)
+        except (CSVWriteError, PermissionError) as e:
+            return False, str(e)
+
+    def clean_data(self, entries: List[Any]) -> List[Dict[str, Any]]:
+        new_entries = []
+        for entry in entries:
+            new_entries.append(self.serialize_data_for_csv(entry))
+
+        return new_entries
+
+    # pylint: disable=no-self-use
+    def serialize_data_for_csv(self, entry: Any) -> Dict[str, Any]:
```

Comment:
Since you have the no-self-use warning you can just make this into a static method.

```suggestion
    @staticmethod
    def serialize_data_for_csv(entry: Any) -> Dict[str, Any]:
```